### DISCUSSION TITLE
Fix constructor in `PRBProxy.sol` contract

### DISCUSSION THREAD

**Discussion by luislucena16** [PaulRBerg/prb-proxy#192] [Q&A] [answered]
looking a bit more at the contract, I see an unchecked return value in the constructor inside [`PRBPorxy. sol`](https://github.com/PaulRBerg/prb-proxy/blob/43a87c48d26dbabe9fe8eb7ba8113044facb5159/src/PRBProxy.sol#L39C1-L47C1):

```solidity
    constructor() {
        registry = IPRBProxyRegistry(msg.sender);
        (address owner_, address target, bytes memory data) = registry.constructorParams();
        owner = owner_;
        if (target != address(0)) {
            _execute(target, data);
        }
    }
```

having in mind that [`_execute`](https://github.com/PaulRBerg/prb-proxy/blob/43a87c48d26dbabe9fe8eb7ba8113044facb5159/src/PRBProxy.sol#L90C1-L101C6) function return a `bytes` value, but this process is not checked, a new `error` would also be added to revert if it fails!

**Comment by PaulRBerg** [comment, marked as answer]
The design is intentional. The point is to let `_execute` handle the revert.

> a new error would also be added to revert if it fails!

What is your suggestion, more specifically? A code snippet would be helpful

**Reply by luislucena16** [reply]
ok, now I understand! the goal is to take the error in `_execute`, I thought you were looking to do it there directly, something like this:

```solidity
constructor() {
    registry = IPRBProxyRegistry(msg.sender);
    (address owner_, address target, bytes memory data) = registry.constructorParams();
    owner = owner_;
    if (target != address(0)) {
        bytes memory response = _execute(target, data);
        if (response.length == 0) {
            revert ExecutionFailed();
        }
    }
}
```
